packages/jest-core/src/cli/index.ts

```diff
@@ -204,11 +213,13 @@ const _run10000 = async (
     };
   }
 
+  performance.mark('jest/buildContextsAndHasteMaps:start');
   const {contexts, hasteMapInstances} = await buildContextsAndHasteMaps(
     configs,
     globalConfig,
     outputStream,
   );
+  performance.mark('jest/buildContextsAndHasteMaps:end');
```

Comment:
would using something like https://nodejs.org/api/perf_hooks.html#performancetimerifyfn-options make sense? both here and other places with a `start` & `end` pair

Reply:
I did consider `timerify`, but it's limited - there's no way to pass `detail`, unlike with marks (I used that here for `numTests`), and the only label on the entry is the literal name of the function, which isn't necessarily descriptive out of context. It's not interchangeable with `mark` either, because it requires a `PerformanceObserver` and creates different entry types.

One thing I like about being able to specify the labels is being able to namespace them to Jest, which is important if you've got other instrumented code in the same process (if, e.g, Babel introduced instrumentation, we'd want to be able to distinguish it from Jests, and at Meta we run Jest programmatically in a wrapper that has its own instrumentation).


Another alternative I looked at is `measure`, but it pretty much just changes the syntax of the end marker:
```js
const fooStartMarker = performance.mark('foo:start');
// ...foo
performance.measure('foo', fooStartMarker); // OR performance.measure('foo', 'foo:start');
```
vs
```js
performance.mark('foo:start');
// ...foo
performance.mark('foo:end');
```

..I'm not sure the former is any cleaner. Also, a consumer can trivially create measures from marks if that's what they want, whereas the reverse is harder.

So TL;DR IMO `mark` optimises for flexibility and simplicity. Admittedly it isn't super clean, but I wouldn't anticipate littering the whole codebase or even much more than what's in this PR - if we did end up adding lots more markers, we could reduce some duplication and strengthen typing with an abstraction (`require('jest-instrumentation').startMark('foo')` or similar).

Reply:
Cool, thanks for the thorough response 👍